streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -109,27 +111,36 @@ public void register(StateStore store, StateRestoreCallback stateRestoreCallback
             throw new IllegalArgumentException("Store " + store.name() + " has already been registered.");
 
         // check that the underlying change log topic exist or not
-        if (restoreConsumer.listTopics().containsKey(store.name())) {
-            boolean partitionNotFound = true;
-            for (PartitionInfo partitionInfo : restoreConsumer.partitionsFor(store.name())) {
+        String topic = store.name() + STATE_CHANGELOG_TOPIC_SUFFIX;
```

Comment:
Why adding a suffix? Is there any problem if the topic name is equal to the store name?
This breaks ktable.